go/vt/schemadiff/table_test.go

```diff
@@ -438,6 +438,20 @@ func TestCreateTableDiff(t *testing.T) {
 			diff:  "alter table t1 add fulltext key name_ft (`name`)",
 			cdiff: "ALTER TABLE `t1` ADD FULLTEXT KEY `name_ft` (`name`)",
 		},
+		{
+			name:  "add one fulltext key with explicit parser",
+			from:  "create table t1 (id int primary key, name tinytext not null)",
+			to:    "create table t1 (id int primary key, name tinytext not null, fulltext key name_ft(name) with parser ngram)",
+			diff:  "alter table t1 add fulltext key name_ft (`name`) with parser ngram",
+			cdiff: "ALTER TABLE `t1` ADD FULLTEXT KEY `name_ft` (`name`) WITH PARSER NGRAM",
```

Comment:
I think we also want to / need to change the canonical output format of this. It looks like the parser formatter always upcases this. I think we should add something like the following as well:

```patch
diff --git i/go/vt/sqlparser/ast_format.go w/go/vt/sqlparser/ast_format.go
index e94a1b24ab..65756a46a9 100644
--- i/go/vt/sqlparser/ast_format.go
+++ w/go/vt/sqlparser/ast_format.go
@@ -826,7 +826,7 @@ func (idx *IndexDefinition) Format(buf *TrackedBuffer) {
        for _, opt := range idx.Options {
                buf.astPrintf(idx, " %s", opt.Name)
                if opt.String != "" {
-                       buf.astPrintf(idx, " %s", opt.String)
+                       buf.astPrintf(idx, " %#s", opt.String)
                } else if opt.Value != nil {
                        buf.astPrintf(idx, " %v", opt.Value)
                }
diff --git i/go/vt/sqlparser/tracked_buffer_test.go w/go/vt/sqlparser/tracked_buffer_test.go
index 02ea192a5d..614a215c0e 100644
--- i/go/vt/sqlparser/tracked_buffer_test.go
+++ w/go/vt/sqlparser/tracked_buffer_test.go
@@ -220,6 +220,10 @@ func TestCanonicalOutput(t *testing.T) {
                        "select char(77, 121, 83, 81, '76' using utf8mb4) from dual",
                        "SELECT CHAR(77, 121, 83, 81, '76' USING utf8mb4) FROM `dual`",
                },
+               {
+                       "create table t1 (id int primary key, name tinytext not null, fulltext key name_ft(name) with parser ngram)",
+                       "CREATE TABLE `t1` (\n\t`id` int PRIMARY KEY,\n\t`name` tinytext NOT NULL,\n\tFULLTEXT KEY `name_ft` (`name`) WITH PARSER ngram\n)",
+               },
        }
 
        for _, tc := range testcases {
```

Reply:
added